central/networkgraph/service/service_impl.go

```diff
@@ -160,6 +161,27 @@ func (s *serviceImpl) GetExternalNetworkFlows(ctx context.Context, request *v1.G
 	}, nil
 }
 
+func (s *serviceImpl) GetFlowsByEntity(ctx context.Context, request *v1.GetFlowsByEntityRequest) (*v1.GetFlowsByEntityResponse, error) {
+	entity, found, err := s.entityDS.GetEntity(ctx, request.GetEntityId())
+	if err != nil || !found {
+		return nil, err
+	}
```

Comment:
I am assuming that this check is intended to save some resources in case the ID doesn't exist at all. I note that it is possible that the check passes, but the entity doesn't belong to the specified cluster afterwards.